Build failure: llama-cpp-vulkan

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

1. Create a `llama-cpp.nix` file and fill it with the contents of the [`package.nix`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/ll/llama-cpp/package.nix) for `llama-cpp` in the unstable branch
2. Add the local package via `(callPackage ./llama-cpp.nix { vulkanSupport = true; })`
3. Rebuild the system configuration
4. Receive a build error that states `fatal error: spirv/unified1/spirv.hpp: No such file or directory`
5. Add `spirv-headers` to the build inputs inside `llama-cpp.nix`
6. Rebuild the system configuration
7. `llama-cpp` now builds without errors

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: Cannot build '/nix/store/2b6pm2wl332sdf0dz03vwff4azy3n83p-llama-cpp-8983.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/pvp92fpbm6zxmkqh6jfj7cf4bsgmwjsv-llama-cpp-8983-dev
         /nix/store/vnwimvh9z8klb5ps8q2j2g9scrzzn8by-llama-cpp-8983
       Last 25 log lines:
       > [200/624] Generate vulkan shaders for mul_mm.comp
       > [201/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/abs.comp.cpp.o
       > [202/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/acc.comp.cpp.o
       > [203/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/add1.comp.cpp.o
       > [204/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/add_id.comp.cpp.o
       > [205/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/arange.comp.cpp.o
       > [206/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/argmax.comp.cpp.o
       > [207/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/add.comp.cpp.o
       > [208/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/argsort.comp.cpp.o
       > [209/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/argsort_large.comp.cpp.o
       > [210/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ceil.comp.cpp.o
       > [211/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/clamp.comp.cpp.o
       > [212/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/contig_copy.comp.cpp.o
       > [213/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/concat.comp.cpp.o
       > [214/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o
       > FAILED: [code=1] ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o
       > /nix/store/kilyfvfmh9x9rlaj47ph1916lbc1pnsr-gcc-wrapper-14.3.0/bin/g++ -DGGML_BACKEND_BUILD -DGGML_BACKEND_SHARED -DGGML_SCHED_MAX_COPIES=4 -DGGML_SHARED -DGGML_VULKAN_BFLOAT16_GLSLC_SUPPORT -DGGML_VULKAN_COOPMAT2_GLSLC_SUPPORT -DGGML_VULKAN_COOPMAT_GLSLC_SUPPORT -DGGML_VULKAN_INTEGER_DOT_GLSLC_SUPPORT -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -Dggml_vulkan_EXPORTS -I/build/source/ggml/src/ggml-vulkan/.. -I/build/source/build/ggml/src/ggml-vulkan -I/build/source/ggml/src/../include -O3 -DNDEBUG -std=gnu++17 -fPIC -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-array-bounds -Wextra-semi -MD -MT ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o -MF ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o.d -o ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/ggml-vulkan.cpp.o -c /build/source/ggml/src/ggml-vulkan/ggml-vulkan.cpp
       > /build/source/ggml/src/ggml-vulkan/ggml-vulkan.cpp:35:14: fatal error: spirv/unified1/spirv.hpp: No such file or directory
       >    35 | #    include <spirv/unified1/spirv.hpp>
       >       |              ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > [215/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/conv2d_dw.comp.cpp.o
       > [216/624] Building CXX object ggml/src/ggml-vulkan/CMakeFiles/ggml-vulkan.dir/conv2d_mm.comp.cpp.o
       > [217/624] Building CXX object vendor/cpp-httplib/CMakeFiles/cpp-httplib.dir/httplib.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run:
         nix log /nix/store/2b6pm2wl332sdf0dz03vwff4azy3n83p-llama-cpp-8983.drv
error: Cannot build '/nix/store/vk9n9sg33li79v4sb29x2x0mamxhixgl-system-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/82glmjq4s01yikfdj5ll8xh3xyc13ak3-system-path
error: Cannot build '/nix/store/9yjjmapyjfpw0sg4hlgzv7srv409ab3n-nixos-system-nixos-25.11.20260429.755f5aa.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/jsgkwk0v8vszcgi81p5yhjqbg1lm5vk6-nixos-system-nixos-25.11.20260429.755f5aa
Command 'nix --extra-experimental-features 'nix-command flakes' build --print-out-paths '/etc/nixos#nixosConfigurations."nixos".config.system.build.toplevel' --no-link' returned non-zero exit status 1.
```

### Additional context

Hydra has `spirv-headers` in its build dependencies for `llama-cpp-vulkan`, despite being absent in the [`package.nix`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/ll/llama-cpp/package.nix) file in nixpkgs, which allows you to download the cached version, but causes a local build to fail.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 7.0.0, NixOS, 25.11 (Xantusia), 25.11.20260421.10e7ad5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.4`
 - nixpkgs: `/nix/store/ds6zw8983xm46y10dq7aswqkgmry968q-source`

### Notify maintainers

@booxter
@dit7ya
@philiptaron
@xddxdd
@yuannan

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ll/llama-cpp/package.nix
+++ b/pkgs/by-name/ll/llama-cpp/package.nix
@@ -40,6 +40,7 @@
   shaderc,
   vulkan-headers,
   vulkan-loader,
+  spirv-headers,
   ninja,
 }:
 
@@ -78,7 +79,7 @@ let
 in
 effectiveStdenv.mkDerivation (finalAttrs: {
   pname = "llama-cpp";
-  version = "8983";
+  version = "9080";
 
   outputs = [
     "out"
@@ -89,7 +90,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     owner = "ggml-org";
     repo = "llama.cpp";
     tag = "b${finalAttrs.version}";
-    hash = "sha256-v73SfPbzzLBNWyan1dsAnDc2dfSem36zyeGLen5t4KI=";
+    hash = "sha256-IF2Whu9VLxUbL+oLkQeYERfp8s3RdQBUyqsAsruv60M=";
     leaveDotGit = true;
     postFetch = ''
       git -C "$out" rev-parse --short HEAD > $out/COMMIT
@@ -106,6 +107,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     nodejs
     npmHooks.npmConfigHook
     pkg-config
+    spirv-headers
   ]
   ++ optionals cudaSupport [
     cudaPackages.cuda_nvcc
@@ -121,7 +123,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     ++ [ openssl ];
 
   npmRoot = "tools/server/webui";
-  npmDepsHash = "sha256-iYJB0z2YHG8OzEA9EwHUZrDa5obr5m2sbnIH+of28o0=";
+  npmDepsHash = "sha256-cV3noOyKmst9vfxyvkCNhihPgwfVGhmPPT4UMloeWZM=";
   npmDeps = fetchNpmDeps {
     name = "${finalAttrs.pname}-${finalAttrs.version}-npm-deps";
     inherit (finalAttrs) src patches;

--- a/pkgs/by-name/ll/llama-cpp/package.nix
+++ b/pkgs/by-name/ll/llama-cpp/package.nix
@@ -40,6 +40,7 @@
   shaderc,
   vulkan-headers,
   vulkan-loader,
+  spirv-headers,
   ninja,
 }:
 
@@ -78,7 +79,7 @@ let
 in
 effectiveStdenv.mkDerivation (finalAttrs: {
   pname = "llama-cpp";
-  version = "8983";
+  version = "9006";
 
   outputs = [
     "out"
@@ -89,7 +90,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     owner = "ggml-org";
     repo = "llama.cpp";
     tag = "b${finalAttrs.version}";
-    hash = "sha256-v73SfPbzzLBNWyan1dsAnDc2dfSem36zyeGLen5t4KI=";
+    hash = "sha256-2OO4wSMI8LK8NkvchZbcujDJcjk/itEAgwXa0icbVwU=";
     leaveDotGit = true;
     postFetch = ''
       git -C "$out" rev-parse --short HEAD > $out/COMMIT
@@ -106,6 +107,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     nodejs
     npmHooks.npmConfigHook
     pkg-config
+    spirv-headers
   ]
   ++ optionals cudaSupport [
     cudaPackages.cuda_nvcc
@@ -121,7 +123,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
     ++ [ openssl ];
 
   npmRoot = "tools/server/webui";
-  npmDepsHash = "sha256-iYJB0z2YHG8OzEA9EwHUZrDa5obr5m2sbnIH+of28o0=";
+  npmDepsHash = "sha256-k62LIbyY2DXvs7XXbX0lNPiYxuYzeJUyQtS4eA+68f8=";
   npmDeps = fetchNpmDeps {
     name = "${finalAttrs.pname}-${finalAttrs.version}-npm-deps";
     inherit (finalAttrs) src patches;

